File: VB/WindowsApplication1/Custom style/XtraTabProgressPage.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraTab.Registrator
Imports DevExpress.XtraTab.Drawing
Imports DevExpress.XtraTab

Namespace WindowsApplication1
	Public Class XtraTabProgressPage
		Inherits XtraTabPage

		Public Sub New()

		End Sub

		Private _Progress As Integer
		Public Property Progress() As Integer
			Get
				Return _Progress
			End Get
			Set(ByVal value As Integer)
				If value < 0 Then
					value = 0
				End If
				If value > 100 Then
					value = 100
				End If
				_Progress = value
				OnProgressChanged()
			End Set
		End Property
		Private Sub OnProgressChanged()
			Parent.Refresh()
		End Sub

		Public Overrides Property Text() As String
			Get
				Return String.Format("{0} ({1}%) ", MyBase.Text, Progress)
			End Get
			Set(ByVal value As String)
				MyBase.Text = value
			End Set
		End Property
	End Class
End Namespace

File: CS/WindowsApplication1/Custom style/XtraTabProgressPage.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraTab.Registrator;
using DevExpress.XtraTab.Drawing;
using DevExpress.XtraTab;

namespace WindowsApplication1
{
    public class XtraTabProgressPage : XtraTabPage
    {

        public XtraTabProgressPage()
        {

        }

        private int _Progress;
        public int Progress
        {
            get { return _Progress; }
            set
            {
                if (value < 0)
                    value = 0;
                if (value > 100)
                    value = 100;
                _Progress = value;
                OnProgressChanged();
            }
        }
        private void OnProgressChanged()
        {
            Parent.Refresh();
        }

        public override string Text
        {
            get
            {
                return String.Format("{0} ({1}%) ", base.Text, Progress);
            }
            set
            {
                base.Text = value;
            }
        }
    }
}
